luda-editor/client/src/pages/sequence_edit_page/components/name_quick_slot_modal/mod.rs

```diff
@@ -0,0 +1,205 @@
+use crate::{color, pages::sequence_edit_page::atom::NAME_QUICK_SLOT};
+use namui::{prelude::*, text_input::Style};
+use namui_prebuilt::{button::TextButton, simple_rect, table::hooks::*, typography};
+
+const MODAL_MAX_WH: Wh<Px> = Wh {
+    width: px(512.0),
+    height: px(227.0),
+};
+const TITLE_HEIGHT: Px = px(32.0);
+const ITEM_HEIGHT: Px = px(39.0);
+const ITEM_PADDING: Px = px(8.0);
+const NAME_LABEL_WIDTH: Px = px(64.0);
+const TEXT_INPUT_PADDING: Ltrb<Px> = Ltrb {
+    left: px(4.0),
+    top: px(4.0),
+    right: px(4.0),
+    bottom: px(4.0),
+};
+
+pub enum Event {
+    Close,
+}
+
+#[component]
+pub struct NameQuickSlotModal<'a> {
+    pub wh: Wh<Px>,
+    pub on_event: &'a dyn Fn(Event),
+}
+impl Component for NameQuickSlotModal<'_> {
+    fn render(self, ctx: &RenderCtx) -> RenderDone {
+        let Self { wh, on_event } = self;
+
+        let modal_wh = Wh {
+            width: wh.width.min(MODAL_MAX_WH.width),
+            height: wh.height.min(MODAL_MAX_WH.height),
+        };
+        let modal_xy = ((wh - modal_wh) / 2.0).as_xy();
+        let (name_quick_slot, set_name_quick_slot) = ctx.atom(&NAME_QUICK_SLOT);
+        let text_input_instances = [
+            namui::text_input::TextInputInstance::new(ctx),
+            namui::text_input::TextInputInstance::new(ctx),
+            namui::text_input::TextInputInstance::new(ctx),
+            namui::text_input::TextInputInstance::new(ctx),
+            namui::text_input::TextInputInstance::new(ctx),
+        ];
+
+        let title_cell = fixed(TITLE_HEIGHT, |wh, ctx| {
+            horizontal([
+                ratio(1, |_, _| {}),
+                fixed(wh.height, |wh, ctx| {
+                    ctx.add(TextButton {
+                        rect: wh.to_rect(),
+                        text: "X",
+                        text_color: color::STROKE_NORMAL,
+                        stroke_color: color::STROKE_NORMAL,
+                        stroke_width: 1.px(),
+                        fill_color: Color::TRANSPARENT,
+                        mouse_buttons: vec![MouseButton::Left],
+                        on_mouse_up_in: Box::new(|_| {
```

Comment:
Change this not to boxed

Reply:
I already opened issue about `namui_prebuilt`'s button series. #700
I think it should be handled on that issue because it's not tiny job. It's quite big.